cinema/views.py

```diff
@@ -56,3 +90,39 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        date = self.request.query_params.get("date")
+        movies = self.request.query_params.get("movie")
+
+        queryset = self.queryset
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movies:
+            movies_ids = [int(str_id) for str_id in movies.split(",")]
+            queryset = queryset.filter(movie__id__in=movies_ids)
+
+        if self.action == ("list", "retrieve"):
```

Comment:
The condition `if self.action == ('list', 'retrieve'):` is incorrect. Use `if self.action in ['list', 'retrieve']:` to properly check if the action is either 'list' or 'retrieve'.